### DISCUSSION TITLE
Make links to 'page-not-created' articles inert?

### DISCUSSION THREAD

**Discussion by robinmetral** [mdn/mdn-community#300] [Platform] [answered]
Currently, yari transforms links to pages that aren't created yet by [adding `class="page-not-created"`](https://github.com/mdn/yari/blob/2720d1f9998be94428a822dcc06946d6a53879d0/kumascript/src/api/web.ts#L180).

This class [adds styles](https://github.com/mdn/yari/blob/2720d1f9998be94428a822dcc06946d6a53879d0/client/src/document/index.scss#L584-L594) that show sighted users that the page isn't working.

I think this approach has several issues:

- For sighted users, it doesn't seem necessary for the link to actually have an `href`. When clicking the link with `page-not-created` styles, the user just lands on a 404 (it might make more sense if the target page presented a [stub](https://en.wikipedia.org/wiki/Wikipedia:Stub) à la Wikipedia, or steps to contribute the page to MDN).
- For sighted pointer device users, the cursor changes to a "🚫". So the fact that the link isn't inert is surprising.
- For non-sighted or low-vision users, the `page-not-created` styles and [title attribute](https://www.tpgi.com/using-the-html-title-attribute/) are not sufficient to indicate that the target page doesn't exist.

The latter point itself seems reason enough to revisit this.

A solution could be to remove links semantics (to prevent confusion for users visiting them) and to explicitly add something like `(page not created)` next to them (with optionally a link to contribution guidelines).

This is intended to start the discussion around the topic. There might be other things to consider (for example, I've seen that there is also logic for showing UI when the target page exists but only in another language).

When we settle on a potential solution, I'd be happy to help with contributing the implementation.

**Comment by schalkneethling** [comment]
Thank you for starting the discussion, @robinmetral. /cc @mdn/core-dev

**Comment by bsmth** [comment, marked as answer]
Thanks for reporting. The first (and second?) points no longer apply since https://github.com/mdn/rari/commit/d5329699daade5b0bd62f17cecb5e394179efff3 where the `href` is removed in favor of a data attribute, so there are no 404ing internal links like this. 

For the third point, does it still apply? The article seems to suggest that making `page not created` part of the link text would help in these cases, is that correct?

__edit:__ for convenience, there are two example 404s here (that may not be fixed right away) to see current implementation https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#trusted_types_api

**Reply by robinmetral** [reply]
Yup these changes seem to fix most things :tada:

The only thing missing is that putting information in `title` ("The documentation about this has not yet been written; please consider contributing!") is generally a bad practice: it's completely unavailable to some users of assistive technology and mobile users. This would probably be flagged in an acessibility audit.

But since this is minor compared to the 404-ing links,  I'll mark the discussion as resolved. Feel free to open a new thread/issue specifically to make the `title` content accessible :slightly_smiling_face:
